src/download.ts

```diff
@@ -77,16 +79,44 @@ export type ResolveHnsResponse = {
 };
 
 export const defaultDownloadOptions = {
-  ...defaultOptions("/"),
+  ...defaultBaseOptions,
+  endpointDownload: "/",
+  download: false,
+  noResponseMetadata: false,
+  path: undefined,
+  query: undefined,
+  subdomain: false,
 };
 const defaultDownloadHnsOptions = {
-  ...defaultOptions("/hns"),
+  ...defaultDownloadOptions,
+  endpointDownloadHns: "hns",
   hnsSubdomain: "hns",
 };
 const defaultResolveHnsOptions = {
-  ...defaultOptions("/hnsres"),
+  ...defaultBaseOptions,
+  endpointDownloadHnsres: "hnsres",
 };
 
+/**
+ * Extract only the download custom options from the given options.
+ *
+ * @param opts - The given options.
+ * @returns - The extracted download custom options.
+ */
+export function extractDownloadOptions(opts: Record<string, unknown>): CustomDownloadOptions {
+  const baseOpts = extractBaseCustomOptions(opts);
+  const downloadOpts = (({ endpointDownload, download, noResponseMetadata, path, query, subdomain }) => ({
+    endpointDownload,
+    download,
+    noResponseMetadata,
+    path,
+    query,
+    subdomain,
+  }))(opts);
+  // @ts-expect-error - We can't ensure the correct types here.
+  return { ...baseOpts, ...downloadOpts };
```

Comment:
couldn't you do this using basic destructuring? 
If you have an object like this:
```
const opts = {
    "name": "foo",
    "path": "bar",
    "size": "baz",
    "date": "hello",
}
```
you can destructure a subset by doing:
```
const { name, path, ...rest } = opts;
const subset = {name, path};
return subset, rest; 
```
^ very much pseudo code but the idea should be clear

Reply:
That's basically what this did :P Not relevant anymore however.